extensions/standard-processors/processors/SplitJson.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "SplitJson.h"
+
+#include <unordered_map>
+
+#include "core/ProcessSession.h"
+#include "core/ProcessContext.h"
+#include "core/Resource.h"
+#include "utils/ProcessorConfigUtils.h"
+#include "utils/Id.h"
+
+#include "jsoncons_ext/jsonpath/jsonpath.hpp"
+
+namespace org::apache::nifi::minifi::processors {
+
+void SplitJson::initialize() {
+  setSupportedProperties(Properties);
+  setSupportedRelationships(Relationships);
+}
+
+void SplitJson::onSchedule(core::ProcessContext& context, core::ProcessSessionFactory&) {
+  json_path_expression_ = utils::parseProperty(context, SplitJson::JsonPathExpression);
+  null_value_representation_ = utils::parseEnumProperty<split_json::NullValueRepresentationOption>(context, SplitJson::NullValueRepresentation);
+}
+
+std::optional<jsoncons::json> SplitJson::queryArrayUsingJsonPath(core::ProcessSession& session, const std::shared_ptr<core::FlowFile>& flow_file) const {
+  const auto json_string = to_string(session.readBuffer(flow_file));
+  if (json_string.empty()) {
+    logger_->log_error("FlowFile content is empty, transferring to Failure relationship");
```

Comment:
```suggestion
    logger_->log_error("FlowFile content is empty, transferring to the 'failure' relationship");
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1980/commits/0aa867f69a964c3d3f0a1562750bd81218fc6434